herbert_core/utilities/general/accumulate_array_to_buffer.m

```diff
@@ -0,0 +1,79 @@
+function [buffer, nel] = accumulate_array_to_buffer (buffer, nel, vals)
+% Append a set of values to a buffer array, doubling its number of elements
+% if there is not enough space left in the buffer in anticipation of further
+% accumulation.
+%
+% Use this rather than the Matlab intrinsic cat function to reduce the amount of
+% memory reallocation should there a large number of accumulations.
+%
+% Append to an exiting array:
+%   >> [buffer, nel] = accumulate_array_to_buffer (buffer, nel, appendvals)
+%
+% To create a buffer:
+%   >> [buffer, nel] = accumulate_array_to_buffer (buffer_size)
+%
+%       Merely a simple utility: creates an array size buffer_size filled with
+%       NaNs and sets nel=0)
```

Comment:
while this is implied, worth saying that to begin a new array, use 2 calls; 1 to create a new empty buffer, one to accumulate the expected contents in that array. Might be an idea to combine them

Reply:
Done